ngraph/core/reference/include/ngraph/runtime/reference/logical_reduction.hpp

```diff
@@ -17,53 +18,75 @@ namespace ngraph
         {
             static inline void reduce_logical_and(const char* arg,
                                                   char* out,
-                                                  const Shape& input_shape,
-                                                  const AxisSet& reduction_axes,
-                                                  bool keep_dims)
+                                                  const Shape& in_shape,
+                                                  const AxisSet& reduction_axes)
             {
-                CoordinateTransform output_transform(
-                    reduce(input_shape, reduction_axes, keep_dims));
+                constexpr bool dont_keep_dims_in_output = false;
```

Comment:
The name of this variable is a little confusing to me. The consuming function `reduce()` takes it's value as `keep_dims` parameter. Should this variable be named the same way?

Reply:
The prev implementation actually used `keep_dims` as this value could be `true` or `false`. But for index calculations we do not really need `keep_dims` attribute. Since resulting dimensions are 1, it does not affect the index. For this reason I created a constant expression as false to indicate that in ref impl we do not keep reduced dimensions in output. The name reflects the purpose of the variable. This change was suggested and merged in other ref implementations as ReduceSum and ReduceProd. Here original [thread](https://github.com/openvinotoolkit/openvino/pull/5728#discussion_r639640796).
I would rather keep it as it is now.